### PR TITLE
Feature/11596 cloud watch logs data protection

### PR DESCRIPTION
**Pull request by timo92**
Closes: #11596 

Note that I was unable to run some tests especially integration-tests locally even after following the steps outlined in https://github.com/serverless/serverless/blob/main/CONTRIBUTING.md#code-style

### CODE REVIEW

**Comment by medikoo** [comment]
> Note that I was unable to run some tests especially integration-tests locally even after following the steps outlined in https://github.com/serverless/serverless/blob/main/CONTRIBUTING.md#code-style

Documentation towards testing can be found at: https://github.com/serverless/serverless/tree/main/test#how-to-run-organize-and-write-tests

**Comment by medikoo** [comment]
@timo92 Let us know, once PR is ready

**Comment by timo92** [comment]
Managed to run `npm run test` successfully (had to delete my AWS_SDK_LOAD_CONFIG env variable).
PR is ready for review @medikoo

File: test/unit/lib/plugins/aws/package/compile/events/websockets/lib/stage.test.js [hunk 1]
```diff
@@ -149,6 +149,29 @@ describe('#compileStage()', () => {
...
+
+        expect(resources[logGroupLogicalId]).to.deep.equal({
+          Type: 'AWS::Logs::LogGroup',
+          Properties: {
+            LogGroupName: '/aws/websocket/my-service-dev',
+            DataProtectionPolicy: policy,
+          },
+        });
+      });
+    });
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/events/websockets/lib/stage.test.js:173]: Let's configure all tests with `runServerless` (see last `describe` section in this file, it should be put there)

**Review by medikoo — COMMENTED** [review]
Thank you @timo92 that looks very good!

File: test/unit/lib/plugins/aws/package/compile/events/websockets/lib/stage.test.js (hunk 1 above)
Comment (timo92) [reply test/unit/lib/plugins/aws/package/compile/events/websockets/lib/stage.test.js]: Anything else missing? @medikoo

Comment (medikoo) [reply test/unit/lib/plugins/aws/package/compile/events/websockets/lib/stage.test.js]: @timo92 if everything is addressed and you feel it's ready for re-review, please re-request it in top right reviewers box. Thank you!

**Review by medikoo — APPROVED** [review]
Thank you @timo92 that looks great!

**Pull request merged**